### ISSUE TITLE
Arbitrary properties of function object have type any

### ISSUE THREAD

**Issue by dmitry-vsl** [facebook/flow#106]
Search keywords: function, statics

Flow finds no error in the following code:

```
/* @flow */
function noop(){}
console.log(noop.foo/noop.bar);
```

I think it is a bug because both noop.foo and noop.bar are undefined.

**Comment by facebook-github-bot-2** [comment]
Thank you for reporting this issue and appreciate your patience. We've  notified the core team for an update on this issue. We're looking for a response within the next 30 days or the issue may be closed.

**Comment by savvopoulos** [comment]
Would be curious to hear if there's an update on this.

**Comment by StyMaar** [comment]
Any news on this ?

**Comment by nmn** [comment]
@avikchaudhuri Do you think defaulting to `?mixed` instead of `any` will be easy to do for this? That seems to solve the problem.

**Comment by danwang** [comment]
bump

**Comment by joelochlann** [comment]
Just re-activating this thread, by pointing out that this weakness in Flow's type system does lead to real-world bugs, particularly around the use of promises.

For instance:

```javascript
function getPromise(): Promise<boolean> {
  return Promise.resolve(true)
}

// type-checks, works at runtime
getPromise().then(() => true);

// type-checks, fails at runtime
getPromise.then(() => true);
```
https://flow.org/try/#0GYVwdgxgLglg9mABAcwKZQAoCc4FsYDOqAFAJQBci2ehqAPAEZxwA2qAhmAHyIDeAUIkRZ0ILEmr4iAOhEFWANxJQsIVKX4BffvzSYcUkqWlQAFqjDEyiALw8Va0gG5d6SbRPnL1u4gfqnIA

I understand that functions are ultimately objects in JavaScript, but so are arrays, and they have stricter type-checking in Flow. Why not functions too?

```javascript
// type-checks, fails at runtime
({}).blah(() => 'hey')

// type-checks, fails at runtime
(() => {}).blah(() => 'hey')

// doesn't type-check, fails at runtime
[].blah(() => 'hey')
```

Any insights you can give into the decisions around this would be much appreciated!

**Comment by kangax** [comment]
@avikchaudhuri I see you mentioned 5 years ago that this will be fixed. Just pinging to see if there's any progress :D

**Comment by gkz** [comment]
Fixed 0.187
